minio/helpers.py

```diff
@@ -123,21 +166,22 @@ def is_valid_endpoint(endpoint):
     parts = urlsplit(endpoint)
     hostname = parts.hostname
     if hostname is None:
-        raise InvalidEndpointError('endpoint')
+        raise InvalidEndpointError('Hostname cannot be empty.')
 
     if len(hostname) > 255:
-        raise InvalidEndpointError('endpoint')
+        raise InvalidEndpointError('Hostname cannot be greater than 255.')
 
     if hostname[-1] == '.':
         hostname = hostname[:-1]
-    allowed = re.compile(
+
+    allowed_hostname = re.compile(
         "^((?!-)[A-Z\\d-]{1,63}(?<!-)\\.)*((?!-)[A-Z\\d-]{1,63}(?<!-))$",
         re.IGNORECASE)
-    if not allowed.match(hostname):
-        raise InvalidEndpointError('endpoint')
+    if not allowed_hostname.match(hostname):
+        raise InvalidEndpointError('Hostname does not meet URL standards.')
 
     if hostname.endswith('amazonaws.com') and (hostname != 's3.amazonaws.com'):
-        raise InvalidEndpointError('endpoint')
+        raise InvalidEndpointError('Hostnames for Amazon S3 should always be s3.amazonaws.com')
```

Comment:
I feel "Amazon S3 hostname should be s3.amazonaws.com" string is better.  Is it good to have hostname in the exception?